pkg/utils/utils.go

```diff
@@ -70,3 +72,35 @@ func DecompressGZIP(compressedData []byte) (decompressedData []byte, err error)
 
 	return decompressedData, nil
 }
+
+// TODO: check if it's worth trying to compress the data prior to compressing it (e.g: using entropy algorithms)
+// This would save some CPU usage client side
+// TODO: have telemetry data type be passed in as a parameter to further check if we should compress data and which algorithm to use (e.g: deflate or gzip)
+func ShouldCompress(data []byte) (resultData []byte, compression *string, err error) {
+	// 'compression' is inserted as a sql.NullString, hence it is returned as a nullable string
+	var nilStr *string = nil
+	var validStr string = "gzip"
+
+	compressedData, err := CompressGZIP(data)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if len(data) <= len(compressedData) {
+		return data, nilStr, nil
+	}
+
+	return compressedData, &validStr, nil
+}
+
+// TODO: have telemetry data type be passed in as a parameter to further check if we should decompress data and which algorithm to use (e.g: deflate or gzip)
+func ShouldDecompress(data []byte, compression sql.NullString) (resultData []byte, err error) {
```

Comment:
Similarly I would probably change the name to something like DecompressFilter() or DecompressIfNeeded()